core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -242,104 +246,64 @@ class ConstructionAutomation(val cityConstructions: CityConstructions) {
         addChoice(relativeCostEffectiveness, spaceshipPart.name, modifier)
     }
 
-    private fun addOtherBuildingChoice() {
-        val otherBuilding = nonWonders
-            .filter { Automation.allowAutomatedConstruction(civInfo, city, it) }
-            .filterBuildable()
-            .minByOrNull { it.cost } ?: return
-        val modifier = 0.6f
-        addChoice(relativeCostEffectiveness, otherBuilding.name, modifier)
-    }
-
-    private fun getWonderPriority(wonder: Building): Float {
-        // Only start building if we are the city that would complete it the soonest
-        if (wonder.hasUnique(UniqueType.TriggersCulturalVictory)
-                && city == civInfo.cities.minByOrNull {
-                    it.cityConstructions.turnsToConstruction(wonder.name)
-                }!!
-        ) {
-            return 10f
-        }
-        if (wonder.name in buildingsForVictory)
-            return 5f
-        if (civInfo.wantsToFocusOn(Victory.Focus.Culture)
-                // TODO: Moddability
-                && wonder.name in listOf("Sistine Chapel", "Eiffel Tower", "Cristo Redentor", "Neuschwanstein", "Sydney Opera House"))
-            return 3f
-        if (wonder.isStatRelated(Stat.Science)) {
-            if (allTechsAreResearched) return .5f
-            return if (civInfo.wantsToFocusOn(Victory.Focus.Science)) 1.5f
-            else 1.3f
-        }
-        if (wonder.hasUnique(UniqueType.EnablesNuclearWeapons)) {
-            return if (civInfo.wantsToFocusOn(Victory.Focus.Military)) 2f
-            else 1.3f
+    private fun addBuildingChoices() {
+        for (building in buildings.filterBuildable() as Sequence<Building>) {
+            if (building.isWonder && city.isPuppet) continue
+            addChoice(relativeCostEffectiveness, building.name, getValueOfBuilding(building))
         }
-        if (wonder.isStatRelated(Stat.Happiness)) return 1.2f
-        if (wonder.isStatRelated(Stat.Production)) return 1.1f
-        return 1f
     }
 
-    private fun addWondersChoice() {
-        if (!wonders.any()) return
-
-        val highestPriorityWonder = wonders
-            .filter { Automation.allowAutomatedConstruction(civInfo, city, it) }
-            .filterBuildable()
-            .maxByOrNull { getWonderPriority(it as Building) }
-            ?: return
+    fun getValueOfBuilding(building: Building): Float {
+        var value = 0f
+        value = applyBuildingStats(building, value)
+        value = applyMilitaryBuildingValue(building, value)
+        value = applyVictoryBuildingValue(building, value)
+        value = applyOnetimeUniqueBonuses(building, value)
+        return value
+    }
```

Comment:
~~(Sorry, been taking a break from stuff to work on separate projects and generally take a break, finally bad to looking over stuff)~~

Why are we doing `value = function(building, value)`. None of these functions actually use the value that's passed in. We could instead do `value += function(building)`

Reply:
>Sorry, been taking a break from stuff

No problem!

> None of these functions actually use the value that's passed in.

Good point, I think that came from an intermediate idea I had and was left over. I'll clean it up.